include/revng/Pipeline/AnyDiff.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <any>
+
+#include "revng/Pipeline/InvalidationEvent.h"
+#include "revng/Pipeline/TupleTreeInvalidationEvent.h"
+#include "revng/Support/YAMLTraits.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+namespace pipeline {
+class AnyDiffBase {
```

Comment:
What about `GenericTupleTreeDiff`? `AnyDiff` doesn't tell me much.

Reply:
i mean, generic tuple tree diff does not convey much more either. `TypeErasedDiff`?

Reply:
Mh, `TypeErased` is the technique through which you achieve the goal, more than the actual role of the class.
I can't come up with anything better than `Generic`. `Opaque`?

Reply:
yeah but this time around we do care about the technique, we had a bunch of templates Diff<T> and we are making a Diff without the T

Reply:
I don't know, I'm not a fan of having `TypeErased` in the name, we should put it in many other places. The property of being type-erased is already in the `Base` suffix. We can switch to an `Abstract` prefix if you prefer (and love Java).

We're a bit overdiscussing this: go with what you like but make sure we don't have `Any` and we have `TupleTreeDiff` in the name. This is my key point, the rest I'll leave it to you.

Reply:
GlobalDiff? In the end nobody else uses it